### PR TITLE
feat(docs): Process for donating community plugins

### PR DESCRIPTION
**Pull request by robzienert**
AWS & Autodesk have a number of plugins that they've been collaboratively developing and want to see community involvement with, however, they feel their company CLAs will be a hinderance to contributorship. The idea here is that the [`spinnaker-plugins`](https://github.com/spinnaker-plugins) organization can be used as a home for these plugins. Plugin authors will continue to own the maintenance and day-to-day operations of the repository via `write` access, however the TOC will have access to the org to help with membership as needed.

Question: Should plugin authors be given `write` or `admin` access for their plugin repository?

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/plugin-ownership-transfer.md
Comment (pstout) [inline .github/ISSUE_TEMPLATE/plugin-ownership-transfer.md:6]: Can this be a link to the document that has the current TOC membership in it?  That way, we do not need to remember to update this template when TOC membership changes.  If that doesn't work, then consider just putting in a prose description that the PR should be assigned to the current members of the TOC.

File: .github/ISSUE_TEMPLATE/plugin-ownership-transfer.md
Comment (pstout) [inline .github/ISSUE_TEMPLATE/plugin-ownership-transfer.md:19]: Does this structure mean that the author of one plugin has write access to all donated plugins?  Is that acceptable?  Is there any mechanism to restrict it to a particular plugin?

Should we describe what people or organizations should do when an author leaves the plugin project or donating organization?

File: .github/ISSUE_TEMPLATE/plugin-ownership-transfer.md
Comment (demobox) [reply .github/ISSUE_TEMPLATE/plugin-ownership-transfer.md:6]: > Can this be a link to the document that has the current TOC membership in it?

+1, if possible

File: .github/ISSUE_TEMPLATE/plugin-ownership-transfer.md
Comment (demobox) [inline .github/ISSUE_TEMPLATE/plugin-ownership-transfer.md:21]: "obligated" is quite a strong term - are they perhaps "expected" to help maintain the plugins? Trying to understand the level of involvement we're aiming for here

File: .github/ISSUE_TEMPLATE/plugin-ownership-transfer.md
Comment (demobox) [reply .github/ISSUE_TEMPLATE/plugin-ownership-transfer.md:19]: My understanding is that plugin "donors" would only be granted write access to *their* plugin's repository - is that correct?

File: sig-platform/community-donated-plugins.md
Comment (demobox) [inline sig-platform/community-donated-plugins.md:13]: What, if any, license conditions apply to plugins donated here? Is it acceptable for them to have any kind of license whatsoever?

File: sig-platform/community-donated-plugins.md
Comment (demobox) [inline sig-platform/community-donated-plugins.md:3]: Is there any plan to make plugins stored here somehow "easy to find", e.g. to index them somewhere, or perhaps even to make them easily installable in Spinnaker itself?

File: .github/ISSUE_TEMPLATE/plugin-ownership-transfer.md
Comment (ajordens) [reply .github/ISSUE_TEMPLATE/plugin-ownership-transfer.md:19]: That would be my expectation. 

`spinnaker-plugins` is a new GitHub organization and each plugin is simply a repository within. Permissions would be managed per-repository.

File: .github/ISSUE_TEMPLATE/plugin-ownership-transfer.md
Comment (ajordens) [reply .github/ISSUE_TEMPLATE/plugin-ownership-transfer.md:21]: I'm not sure there are or should be any expectations that Spinnaker approvers maintain donated community plugins. 

Worth clarifying but I'm assuming Spinnaker approvers would not automatically inherit similar permissions for the `spinnaker-plugins` organization?

**Comment by ajordens** [comment]
@robzienert To answer your question, I think plugin authors should be given write and not admin.

File: sig-platform/community-donated-plugins.md
Comment (ajordens) [reply sig-platform/community-donated-plugins.md:13]: I think it's fair to require that donated plugins declare a license. 

Given that these plugins are not distributed alongside a Spinnaker release, I'm not sure we need to limit what the actual license is though.

File: .github/ISSUE_TEMPLATE/plugin-ownership-transfer.md
Comment (robzienert) [reply .github/ISSUE_TEMPLATE/plugin-ownership-transfer.md:6]: I can try to change it to reference `spinnaker/toc` - this part of the doc is for configuring GitHub, so it won't be something people see unless they're looking at the file - this  `assignees` block will auto-assign the issue to these members.

File: .github/ISSUE_TEMPLATE/plugin-ownership-transfer.md
Comment (robzienert) [reply .github/ISSUE_TEMPLATE/plugin-ownership-transfer.md:19]: Correct - each plugin will have its own repository within the `spinnaker-plugins` org, and the permissions will be scoped to plugin repositories only. SC and TOC will have access to the org as admins, similar to the spinnaker org.

File: .github/ISSUE_TEMPLATE/plugin-ownership-transfer.md
Comment (robzienert) [reply .github/ISSUE_TEMPLATE/plugin-ownership-transfer.md:21]: My intent with this statement was to convey that the Spinnaker core contributors will _not_ be involved in maintenance of the plugins. The responsibility and duties of code ownership will continue to lie entirely with the original authors. Any word smithing help would be great!

> Worth clarifying but I'm assuming Spinnaker approvers would not automatically inherit similar permissions for the spinnaker-plugins organization?

Good clarification, yes, I'll add that - approvers / reviewers will not have their access applied to plugin repos unless the plugin authors want that.

File: sig-platform/community-donated-plugins.md
Comment (robzienert) [reply sig-platform/community-donated-plugins.md:13]: I was debating dictating licenses, but opted against. I do like adding that they must be licensed in some form, but won't make it a requirement to what it is.

File: sig-platform/community-donated-plugins.md
Comment (robzienert) [reply sig-platform/community-donated-plugins.md:3]: Yes! https://github.com/spinnaker/plugins is the draft for a plugin repository. It's similar in concept to maven central, except that we'll be offering just pointers to artifact storage, rather than storing the artifacts themselves. The community has expressed interest in creating a marketplace at some point. This repository concept makes plugins easy to install, there's some docs on that repo of what it looks like (basically, if you don't pre-download the plugins, the services will download and side-load the plugins at startup - the `sha512sum` is required).

File: .github/ISSUE_TEMPLATE/plugin-ownership-transfer.md
Comment (nimakaviani) [inline .github/ISSUE_TEMPLATE/plugin-ownership-transfer.md:21]:
```suggestion
Spinnaker core contributors are not obligated to help maintain the plugin
```

File: sig-platform/community-donated-plugins.md
Comment (nimakaviani) [inline sig-platform/community-donated-plugins.md:9]: I suppose updates to the list of maintainers will also be handled through the TOC members. do we need a clarifying phrase for it or is it assumed?

**Comment by robzienert** [comment]
Updated from feedback.

File: .github/ISSUE_TEMPLATE/plugin-ownership-transfer.md
Comment (demobox) [reply .github/ISSUE_TEMPLATE/plugin-ownership-transfer.md]: Thoughts on setting expectations even lower here, e.g. "Spinnaker core contributors are not expected to help maintain the plugin" or so?

File: sig-platform/community-donated-plugins.md
Comment (demobox) [reply sig-platform/community-donated-plugins.md]: Ack, thanks for explaining! Is that something we could mention here, or is it not sufficiently ready for that?

File: .github/ISSUE_TEMPLATE/plugin-ownership-transfer.md
Comment (demobox) [reply .github/ISSUE_TEMPLATE/plugin-ownership-transfer.md]: Ack, thanks.

@pstout Does that sufficiently address your concern?

File: .github/ISSUE_TEMPLATE/plugin-ownership-transfer.md
Comment (demobox) [reply .github/ISSUE_TEMPLATE/plugin-ownership-transfer.md]: +1 to `spinnaker/toc` to start with? If we end up with too much noise, we can reduce if needed?

File: sig-platform/community-donated-plugins.md
Comment (demobox) [reply sig-platform/community-donated-plugins.md:14]: In that case, perhaps add something along the lines of "Donated code should/must explicitly declare a license."?

File: .github/ISSUE_TEMPLATE/plugin-ownership-transfer.md
Comment (robzienert) [reply .github/ISSUE_TEMPLATE/plugin-ownership-transfer.md]: +1 I'm just not sure if groups are supported in this assignees block. We'll find out once it's merged!

File: sig-platform/community-donated-plugins.md
Comment (robzienert) [reply sig-platform/community-donated-plugins.md]: It's still something we're working through - probably not for another quarter.

**Review by ezimanyi — APPROVED** [review]

**Review by ajordens — APPROVED** [review]

File: sig-platform/community-donated-plugins.md
Comment (demobox) [reply sig-platform/community-donated-plugins.md]: Clear, thanks!

**Comment by nimakaviani** [comment]
How many approvals do we need before this is merged? looks like there hasn't been any outstanding issues.

**Comment by demobox** [comment]
>How many approvals do we need before this is merged? looks like there hasn't been any outstanding issues.

From what I can see, a number of comments (e.g. [this one](https://github.com/spinnaker/governance/pull/181/files#r495403681) are still open?

File: sig-platform/community-donated-plugins.md
Comment (demobox) [reply sig-platform/community-donated-plugins.md:14]: I see the update now - thanks!

**Review by demobox — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/plugin-ownership-transfer.md
Comment (pstout) [reply .github/ISSUE_TEMPLATE/plugin-ownership-transfer.md]: This covers my question about the scope.  I don't think anybody has addressed my question about how we handle people moving on from working on a plugin or the organization that donated that plugin.  A sentence about who does what, or who is allowed to do what, would be useful down the road.

Comment (pstout) [reply .github/ISSUE_TEMPLATE/plugin-ownership-transfer.md]: Well, if I go look at the document, rather than just looking at this comment thread, I can see that you addressed the second part.  Sorry for the noise.

**Review by pstout — APPROVED** [review]

**Review by link108 — APPROVED** [review]

**Pull request merged**
